packages/@expo/server/src/index.ts

```diff
@@ -157,148 +141,207 @@ export function createRequestHandler(
           continue;
         }
 
-        const url = getRedirectRewriteLocation(request, route);
-        request = new Request(new URL(url, new URL(request.url).origin), request);
-        sanitizedPathname = new URL(request.url, 'http://expo.dev').pathname;
+        // Replace URL and Request with rewrite target
+        const location = getRedirectRewriteLocation(request, route);
+        const rewriteUrl = new URL(location, url.origin);
+        request = new Request(rewriteUrl, request);
+        sanitizedPathname = rewriteUrl.pathname;
```

Comment:
More important: This should probably be updated to align with `deployment-workers`'s `getRedirectRewriteLocation` implementation and to remove the redundant `new URL` stuff here which moved over in that implementation. Main reason for the refactor were some subtle issues, if I recall correctly. Either way, the `deployment-worker`'s changes are more recent than the ones in `@expo/server`, so it's always safer to prefer those

Reply:
Thanks, for the clarification, I've also noticed saddle differences between the impls, that's why I added -> https://github.com/expo/expo/pull/38465/files#diff-4506e5f1387d20f70872d508b314e53a22c808ac9adb6679b65d371cd076934a

I'll update this with the `deployment-workers` impl.